### PR TITLE
Add junit-xml

### PR DESCRIPTION
**Pull request by dfornika**

### CODE REVIEW

**Comment by dfornika** [comment]
Moving `junit-xml` here was suggested by @bgruening [here](https://github.com/bioconda/bioconda-recipes/pull/10495).

@johanneskoester and/or @chapmanb can I add you as maintainers since you contributed to the bioconda recipe for this package?

**Comment by chapmanb** [comment]
Thanks for porting this over to conda-forge. I'd be happy to help maintain it, please do add me.

File: recipes/junit-xml/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/j/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} setup.py install"
```
Comment (jakirkham) [inline recipes/junit-xml/meta.yaml]: We try to [use `pip`]( https://conda-forge.org/docs/meta.html#use-pip ) these days when possible.

File: recipes/junit-xml/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - python
+    - six
+
+test:
+  imports:
+    - junit_xml
+
+about:
+  home: https://github.com/kyrus/python-junit-xml
+  license: MIT
```
Comment (jakirkham) [inline recipes/junit-xml/meta.yaml]: Please add `license_file: LICENSE.txt` to package the license file.

File: recipes/junit-xml/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} setup.py install"
+
+requirements:
+  host:
+    - python
+    - setuptools
+    - six
```
Comment (jakirkham) [inline recipes/junit-xml/meta.yaml]: Is `six` needed at build time or only at run time?

File: recipes/junit-xml/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{% set name = "junit-xml" %}
+{% set version = "1.8" %}
+{% set sha256 = "602f1c480a19d64edb452bf7632f76b5f2cb92c1938c6e071dcda8ff9541dc21 " %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/j/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/junit-xml/meta.yaml]: `source/j/` -> `source/{{ name[0] }}/`.

**Comment by jakirkham** [comment]
Looks pretty good. Add a few comments above. Please let us know if you have questions.

**Comment by dfornika** [comment]
I don't understand why it would say that it's installing `pip`, then say `No module named pip`

```
The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-h470a237_2     conda-forge
    ca-certificates: 2018.8.13-ha4d7672_0 conda-forge
    certifi:         2018.8.13-py36_0     conda-forge
    libffi:          3.2.1-hfc679d8_4     conda-forge
    libgcc-ng:       7.2.0-hdf63c60_3     conda-forge
    libstdcxx-ng:    7.2.0-hdf63c60_3     conda-forge
    ncurses:         6.1-hfc679d8_1       conda-forge
    openssl:         1.0.2o-h470a237_1    conda-forge
    pip:             18.0-py36_1          conda-forge
    python:          3.6.6-h5001a0f_0     conda-forge
    readline:        7.0-haf1bffa_1       conda-forge
    setuptools:      40.0.0-py36_1        conda-forge
    sqlite:          3.24.0-h2f33b56_0    conda-forge
    tk:              8.6.8-0              conda-forge
    wheel:           0.31.1-py36_1        conda-forge
    xz:              5.2.4-h470a237_1     conda-forge
    zlib:            1.2.11-h470a237_3    conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Source cache directory is: /home/conda/feedstock_root/build_artifacts/src_cache
Downloading source to cache: junit-xml-1.8_602f1c480a.tar.gz
Downloading https://pypi.io/packages/source/j/junit-xml/junit-xml-1.8.tar.gz
Success
Extracting download
source tree in: /home/conda/feedstock_root/build_artifacts/junit-xml_1534870376293/work
/home/conda/feedstock_root/build_artifacts/junit-xml_1534870376293/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/python: No module named pip
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 83, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 73, in build_all
    conda_build.api.build(list(d.keys()), config=config)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 200, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2209, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1446, in build
    utils.check_call_env(cmd, env=env, cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 310, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 290, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/home/conda/feedstock_root/build_artifacts/junit-xml_1534870376293/work/conda_build.sh']' returned non-zero exit status 1.
Exited with code 1
```

**Comment by mbargull** [comment]
> I don't understand why it would say that it's installing `pip`, then say `No module named pip`

`{{ PYTHON }}` evaluates to the path to the Python interpreter in the `host` environment, so we also need to install `pip` to that environment. Yes, `pip` in technically a build tool so it would make sense to put it in `requirements/build`, but since we have this `python` <-> `pip` dependency, we have to use `host`.

**Comment by dfornika** [comment]
Should the docs be changed?

https://conda-forge.org/docs/meta.html#use-pip

**Comment by mbargull** [comment]
> Should the docs be changed?

Oh, yes, definitely! Can you open an issue for this over at http://github.com/conda-forge/conda-forge.github.io/?

**Comment by bgruening** [comment]
Thanks @dfornika!

**Comment by bgruening** [comment]
Are you doing the follow-up on bioconda and remove the pkg?

**Comment by dfornika** [comment]
I'm not sure how that works. Do I just make a pull request that removes the `recipes/junit-xml` directory?

**Pull request merged**